src/System Application/Test/Performance Profiler/src/ScheduledPerfProfilingTest.Codeunit.al

```diff
@@ -0,0 +1,217 @@
+        PerformanceProfileScheduler."Starting Date-Time" := EndingDateTime - 60000;
+        PerformanceProfileScheduler."Ending Date-Time" := EndingDateTime;
+        asserterror ScheduledPerfProfiler.ValidatePerformanceProfileSchedulerRecord(PerformanceProfileScheduler, ActivityType);
+        Assert.ExpectedError(ProfileHasAlreadyBeenScheduledErr);
+    end;
+
+    [Test]
+    procedure TestValidatePerformanceProfileSchedulerRecordWithNoStartingDate()
+    var
+        PerformanceProfileScheduler: Record "Performance Profile Scheduler";
+        ActivityType: Enum "Activity Type";
+        EndingDateTime: DateTime;
+    begin
+        // [WHEN] we have inserted a new performance profile record
+        ScheduledPerfProfiler.InitializeFields(PerformanceProfileScheduler, ActivityType);
+        EndingDateTime := PerformanceProfileScheduler."Starting Date-Time" + 15 * 60000;
+        PerformanceProfileScheduler."Ending Date-Time" := EndingDateTime;
+        PerformanceProfileScheduler.Insert(true);
+
+        // [THEN] it should not intersect with another that has no ending date
+        Clear(PerformanceProfileScheduler);
+        ScheduledPerfProfiler.InitializeFields(PerformanceProfileScheduler, ActivityType);
+        PerformanceProfileScheduler."Starting Date-Time" := 0DT;
+        PerformanceProfileScheduler."Ending Date-Time" := EndingDateTime;
+        ScheduledPerfProfiler.ValidatePerformanceProfileSchedulerRecord(PerformanceProfileScheduler, ActivityType);
+    end;
+
+    [Test]
+    procedure TestValidatePerformanceProfileSchedulerRecordWithNoEndingDate()
+    var
+        PerformanceProfileScheduler: Record "Performance Profile Scheduler";
+        ActivityType: Enum "Activity Type";
+        EndingDateTime: DateTime;
+    begin
+        // [WHEN] we have inserted a new performance profile record that has no starting date.
+        ScheduledPerfProfiler.InitializeFields(PerformanceProfileScheduler, ActivityType);
+        EndingDateTime := PerformanceProfileScheduler."Starting Date-Time" + 15 * 60000;
+        PerformanceProfileScheduler."Ending Date-Time" := EndingDateTime;
+        PerformanceProfileScheduler.Insert(true);
+
+        // [THEN] it should not intersect with another.
+        Clear(PerformanceProfileScheduler);
+        ScheduledPerfProfiler.InitializeFields(PerformanceProfileScheduler, ActivityType);
+        PerformanceProfileScheduler."Ending Date-Time" := 0DT;
+        ScheduledPerfProfiler.ValidatePerformanceProfileSchedulerRecord(PerformanceProfileScheduler, ActivityType);
+    end;
+
+    [Test]
+    procedure TestUserFilter()
+    var
+        TempPerformanceProfileScheduler: Record "Performance Profile Scheduler" temporary;
+        TempUser: Record User temporary;
+        ActivityType: Enum "Activity Type";
+    begin
+        // [WHEN] we have a user that is just a default user 
+        this.AddTwoUsers(TempUser);
+        if not TempUser.FindSet() then
+            exit;
```

Comment:
This means you try to add AddTwoUsers users, and than if not TempUser.FindSet(), test will be still success, even though it didn't created users and it exited early

Reply:
I would use only FindSet without if condition, so it throws an error if anything. If kind of un-necessary. Since in AddTwoUsers if it doesn't insert it will error, so here it must have Records, if not defenetly you want test to fail.

Reply:
Lets not use if, we should fail the test if the temp user does not exist.

Reply:
Fixed

Reply:
@StefanSosic Btw kudos for this catch and thanks for the review, the test was not correct.